src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -559,11 +707,11 @@ private static void setProperties(String inputFilePath, Configuration module,
 
     private static void setViolations(TestInputConfiguration.Builder inputConfigBuilder,
                                       List<String> lines, boolean useFilteredViolations)
-            throws ClassNotFoundException, CheckstyleException {
+            throws CheckstyleException {
         final List<ModuleInputConfiguration> moduleLists = inputConfigBuilder.getChildrenModules();
         final boolean specifyViolationMessage = moduleLists.size() == 1
```

Comment:
https://github.com/checkstyle/checkstyle/issues/15456
> Define violation messages for all violations

@romani @nrmancuso Why do we only do this for single modules? It does not align with the "all" in the new issue title.

Reply:
I don't remember, we can try to remove in separate issue.

Reply:
Who will do this?

Reply:
I am fine to do this now is we have energy, just pointing that we can do this in few steps, and improve step by step with easy to review PRs.

Reply:
It was added at https://github.com/checkstyle/checkstyle/commit/645262f42ba572b3f3749434f03fd29f74e6cdb6#diff-58cc197c3dfaae5245ca0fb29e5892861aeecff23be676ea61fc95f320b7c318R337

I don't see good reason.
We try to remove, and if no side effects happens, we can keep removal.

Reply:
```
[ERROR] Tests run: 50, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.489 s <<< FAILURE! -- in com.puppycrawl.tools.checkstyle.CheckerTest
[ERROR] com.puppycrawl.tools.checkstyle.CheckerTest.testSetSeverity -- Time elapsed: 0.008 s <<< ERROR!
java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:100)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:106)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:302)
	at java.base/java.util.Objects.checkIndex(Objects.java:365)
	at java.base/java.util.ArrayList.get(ArrayList.java:428)
	at java.base/java.util.Collections$UnmodifiableList.get(Collections.java:1502)
	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.setViolations(InlineConfigParser.java:715)
	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.parseWithXmlHeader(InlineConfigParser.java:443)
	at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.verifyWithInlineXmlConfig(AbstractModuleTestSupport.java:235)
```

There are a lot of failures like this. better to keep this removal in a separate PR

Reply:
```
[INFO] 
[ERROR] Errors: 
[ERROR]   ParenPadCheckTest.testTryWithResourcesAndSuppression:457->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 23 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\checks\whitespace\parenpad\InputParenPadTryWithResourcesAndSuppression.java
[ERROR]   BeforeExecutionExclusionFileFilterTest.testFileExclusion:81->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 17 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filefilters\beforeexecutionexclusionfilefilter\InputBeforeExecutionExclusionFileFilter.java
[ERROR]   SuppressWarningsFilterTest.testDefault:113->verifySuppressedWithParser:140->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 61 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswarningsfilter\InputSuppressWarningsFilter.java
[ERROR]   SuppressWithNearbyCommentFilterTest.tesSuppressByCheckAndMessage:742->verifySuppressedWithParser:374->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 41 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterByCheckAndMessage.java
[ERROR]   SuppressWithNearbyCommentFilterTest.tesSuppressByIdAndMessage:706->verifySuppressedWithParser:374->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 41 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterByIdAndMessage.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testCheckC:258->verifySuppressedWithParser:367->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 46 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterCheckC.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testCheckCpp:281->verifySuppressedWithParser:367->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 46 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterCheckCpp.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testDefault:244->verifySuppressedWithParser:367->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 46 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilter.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testInfluenceFormat:433->verifySuppressedWithParser:367->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 46 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterInfluenceFormat.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testSuppressByCheck:556->verifySuppressedWithParser:374->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 41 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterByCheck.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testSuppressByCheckAndId:638->verifySuppressedWithParser:374->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 41 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterByCheckAndId.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testSuppressByCheckAndNonMatchingId:670->verifySuppressedWithParser:374->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 41 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterByCheckAndNonMatchingId.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testSuppressById:597->verifySuppressedWithParser:374->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 41 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterById.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testUsingNonMatchingVariableMessage:302->verifySuppressedWithParser:367->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 46 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterUsingNonMatchingVariableMessage.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testUsingTagMessageRegexp:514->verifySuppressedWithParser:367->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 46 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterUsingTagMessageRegexp.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testUsingVariableCheckOnNextLine:314->verifySuppressedWithParser:367->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 46 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterUsingVariableCheckOnNextLine.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testUsingVariableCheckOnPreviousLine:326->verifySuppressedWithParser:367->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 46 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterUsingVariableCheckOnPreviousLine.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testUsingVariableMessage:295->verifySuppressedWithParser:367->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 46 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterUsingVariableMessage.java
[ERROR]   SuppressWithNearbyCommentFilterTest.testVariableCheckOnVariableNumberOfLines:347->verifySuppressedWithParser:367->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 46 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithnearbycommentfilter\InputSuppressWithNearbyCommentFilterVariableCheckOnVariableNumberOfLines.java
[ERROR]   SuppressWithPlainTextCommentFilterTest.testSuppressByModuleId:411->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 36 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithplaintextcommentfilter\InputSuppressWithPlainTextCommentFilterSuppressById2.java
[ERROR]   SuppressWithPlainTextCommentFilterTest.testSuppressByModuleIdWithNullModuleId:495->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 36 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithplaintextcommentfilter\InputSuppressWithPlainTextCommentFilterSuppressById5.java
[ERROR]   SuppressWithPlainTextCommentFilterTest.testSuppressedByIdJavadocCheck:520->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 28 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithplaintextcommentfilter\InputSuppressWithPlainTextCommentFilterSuppressByIdJavadocCheck.java
[ERROR]   SuppressWithPlainTextCommentFilterTest.testSuppressionCommentsInJavaScriptFile:192->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 22 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppresswithplaintextcommentfilter\InputSuppressWithPlainTextCommentFilter.js
[ERROR]   SuppressionCommentFilterTest.testArgumentSuppression:255->verifySuppressedWithParser:282->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 45 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressioncommentfilter\InputSuppressionCommentFilter7.java
[ERROR]   SuppressionCommentFilterTest.testCheckC:202->verifySuppressedWithParser:282->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 45 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressioncommentfilter\InputSuppressionCommentFilter3.java
[ERROR]   SuppressionCommentFilterTest.testCheckCpp:214->verifySuppressedWithParser:282->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 45 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressioncommentfilter\InputSuppressionCommentFilter4.java
[ERROR]   SuppressionCommentFilterTest.testDefault:188->verifySuppressedWithParser:282->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 45 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressioncommentfilter\InputSuppressionCommentFilter2.java
[ERROR]   SuppressionCommentFilterTest.testExpansion:271->verifySuppressedWithParser:282->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 45 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressioncommentfilter\InputSuppressionCommentFilter8.java
[ERROR]   SuppressionCommentFilterTest.testMessage:277->verifySuppressedWithParser:282->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 45 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressioncommentfilter\InputSuppressionCommentFilter9.java
[ERROR]   SuppressionCommentFilterTest.testOffFormat:234->verifySuppressedWithParser:282->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 45 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressioncommentfilter\InputSuppressionCommentFilter5.java
[ERROR]   SuppressionCommentFilterTest.testOffFormatCheck:246->verifySuppressedWithParser:282->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 45 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressioncommentfilter\InputSuppressionCommentFilter6.java
[ERROR]   SuppressionCommentFilterTest.testSuppressByCheck:465->verifySuppressedWithParser:289->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 42 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressioncommentfilter\InputSuppressionCommentFilterSuppressById.java
[ERROR]   SuppressionCommentFilterTest.testSuppressByCheckAndId:535->verifySuppressedWithParser:289->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 42 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressioncommentfilter\InputSuppressionCommentFilterSuppressById3.java
[ERROR]   SuppressionCommentFilterTest.testSuppressByCheckAndMessage:599->verifySuppressedWithParser:289->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 42 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressioncommentfilter\InputSuppressionCommentFilterSuppressById5.java
[ERROR]   SuppressionCommentFilterTest.testSuppressById:500->verifySuppressedWithParser:289->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 42 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressioncommentfilter\InputSuppressionCommentFilterSuppressById2.java
[ERROR]   SuppressionCommentFilterTest.testSuppressByIdAndMessage:567->verifySuppressedWithParser:289->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 42 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressioncommentfilter\InputSuppressionCommentFilterSuppressById4.java
[ERROR]   SuppressionFilterTest.testSuppression2:305->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 19 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionfilter\InputSuppressionFilter7.java
[ERROR]   SuppressionFilterTest.testSuppression3:317->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 19 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionfilter\InputSuppressionFilter8.java
[ERROR]   SuppressionFilterTest.testXpathSuppression:288->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 19 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionfilter\InputSuppressionFilter1.java
[ERROR]   SuppressionXpathFilterTest.testAcceptOnNullFile:76->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathfilter\InputSuppressionXpathFilterAcceptOnNullFile.java
[ERROR]   SuppressionXpathFilterTest.testAcceptOne:57->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathfilter\InputSuppressionXpathFilterAcceptOne.java
[ERROR]   SuppressionXpathFilterTest.testAcceptTwo:69->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathfilter\InputSuppressionXpathFilterAcceptTwo.java
[ERROR]   SuppressionXpathFilterTest.testExistingSuppressionFileWithTrueOptional:115->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathfilter\InputSuppressionXpathFilterAcceptWithOptionalTrue.java
[ERROR]   SuppressionXpathFilterTest.testFalseEncodeChar:218->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 17 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathfilter\InputSuppressionXpathFilterEscapeChar.java
[ERROR]   SuppressionXpathFilterTest.testFalseEncodeString:197->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 17 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathfilter\InputSuppressionXpathFilterEscapeString.java
[ERROR]   SuppressionXpathFilterTest.testNonExistentSuppressionFileWithTrueOptional:123->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathfilter\InputSuppressionXpathFilterNonExistentFileWithTrueOptional.java
[ERROR]   SuppressionXpathFilterTest.testReject:134->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathfilter\InputSuppressionXpathFilterReject.java
[ERROR]   SuppressionXpathFilterTest.testXpathSuppression:232->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathfilter\InputSuppressionXpathFilter1.java
[ERROR]   SuppressionXpathFilterTest.testXpathSuppression2:248->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 18 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathfilter\InputSuppressionXpathFilter5.java
[ERROR]   SuppressionXpathFilterTest.testXpathSuppression3:263->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 18 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathfilter\InputSuppressionXpathFilter6.java
[ERROR]   SuppressionXpathFilterTest.testXpathSuppression4:273->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathfilter\InputSuppressionXpathFilter7.java
[ERROR]   SuppressionXpathSingleFilterTest.testAllNullConfiguration:340->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 18 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterAllNullConfiguration.java
[ERROR]   SuppressionXpathSingleFilterTest.testComplexQuery:123->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 27 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterComplexQuery.java
[ERROR]   SuppressionXpathSingleFilterTest.testDecideByCheck:385->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 18 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterDecideByCheck.java
[ERROR]   SuppressionXpathSingleFilterTest.testDecideById:400->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 19 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterDecideById.java
[ERROR]   SuppressionXpathSingleFilterTest.testDecideByIdAndExpression:355->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterDecideByIdAndExpression.java
[ERROR]   SuppressionXpathSingleFilterTest.testDecideByMessage:301->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 28 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterDecideByMessage.java
[ERROR]   SuppressionXpathSingleFilterTest.testDefaultFileProperty:370->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterDefaultFileProperty.java
[ERROR]   SuppressionXpathSingleFilterTest.testMatching:64->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 19 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterMatchingTokenType.java
[ERROR]   SuppressionXpathSingleFilterTest.testMatchingModuleId:246->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterMatchingModuleId.java
[ERROR]   SuppressionXpathSingleFilterTest.testNoQuery:154->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 18 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterNoQuery.java
[ERROR]   SuppressionXpathSingleFilterTest.testNonMatchingChecks:259->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 19 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterNonMatchingCheck.java
[ERROR]   SuppressionXpathSingleFilterTest.testNonMatchingColumnNumber:108->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 23 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterNonMatchingColumnNumber.java
[ERROR]   SuppressionXpathSingleFilterTest.testNonMatchingFileNameModuleIdAndCheck:271->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterNonMatchingFileNameModuleIdAndCheck.java
[ERROR]   SuppressionXpathSingleFilterTest.testNonMatchingFileRegexp:180->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 18 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterNonMatchingFileRegexp.java
[ERROR]   SuppressionXpathSingleFilterTest.testNonMatchingLineNumber:93->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 18 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterNonMatchingLineNumber.java
[ERROR]   SuppressionXpathSingleFilterTest.testNonMatchingModuleId:231->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterNonMatchingModuleId.java
[ERROR]   SuppressionXpathSingleFilterTest.testNonMatchingTokenType:77->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 19 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterNonMatchingTokenType.java
[ERROR]   SuppressionXpathSingleFilterTest.testNullFileName:167->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 18 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterNullFileName.java
[ERROR]   SuppressionXpathSingleFilterTest.testNullModuleIdAndNonMatchingChecks:283->AbstractModuleTestSupport.verifyFilterWithInlineConfigParser:208 » Checkstyle Violation message should be specified on line 20 in D:\CS\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\filters\suppressionxpathsinglefilter\InputSuppressionXpathSingleFilterNullModuleIdAndNonMatchingCheck.java
```

Reply:
I think this can't be resolved because we don't have a [pattern](https://github.com/checkstyle/checkstyle/blob/798e70bb378e31b23c1ba6dbbbea78a0a606868c/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java#L114-L123) `FILTERED_VIOLATION_WITH_EXPLANATION_PATTERN` .

So if we uses 2 modules in a file one of them is the filter then we can't specifiy messages in the filterd violation so this errors can't be resolved.

Reply:
IMO, we should keep this as it is and try to remove it in a separate PR.

Reply:
@mahfouz72 if you think we can do this, then make an issue with the details